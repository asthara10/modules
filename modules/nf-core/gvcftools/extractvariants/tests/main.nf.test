nextflow_process {

    name "Test Process GVCFTOOLS_EXTRACTVARIANTS"
    script "../main.nf"
    process "GVCFTOOLS_EXTRACTVARIANTS"

    tag "modules"
    tag "modules_nfcore"
    tag "gvcftools"
    tag "gvcftools/extractvariants"

    test("test_gvcftools_extractvariants") {


        when {
            process {
                """
                
                input[0] = [
        [ id:'test', single_end:false ], // meta map
        file(params.test_data['homo_sapiens']['illumina']['test_genome_vcf_gz'], checkIfExists: true)
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_gvcftools_extractvariants_uncompressed") {
        when {
            process {
                """
                        input[0] = [
        [ id:'test', single_end:false ], // meta map
        file(params.test_data['homo_sapiens']['illumina']['test_genome_vcf'], checkIfExists: true)
    ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
}
