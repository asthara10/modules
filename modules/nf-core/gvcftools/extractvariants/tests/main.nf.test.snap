{
    "versions": {
        "content": [
            [
                "versions.yml:md5,210e1f24a3b70e240cabf80136720234"
            ]
        ],
        "timestamp": "2023-11-24T13:19:02.591584"
    }
}